docs/rfcs/002-atom-nightly-releases.md

```diff
@@ -0,0 +1,55 @@
+# Atom Nightly Releases
+
+## Status
+
+Proposed
+
+## Summary
+
+This RFC proposes that Atom add a third official release channel which delivers new builds of Atom nightly from the `master` branch.  Nightly releases will allow new improvements to reach users long before a new Stable or Beta release is shipped.  This effort will also give us the opportunity to experiment with new release automation strategies that could eventually be used to speed up the Stable and Beta release cadence.
+
+## Motivation
+
+Atom currently uses a monthly release cycle with staged Stable and Beta releases so that major issues get caught early in Beta before reaching the Stable release.  Because Atom releases updates monthly, this means that a new feature merged into `master` right after a new Atom release could take one month to reach the next Beta and then another month to reach Stable.
+
+This release process works well for delivering stable improvements to users on a regular basis but it results in friction for users who want to try out the latest Atom improvements and provide feedback.  If we deliver a nightly release channel, it will be possible to deliver new features and bug fixes on a regular basis and get valuable feedback to guide our work.
+
+Today, a bleeding-edge user must manually pull Atom's `master` branch and compile their own build.  There is a source of `dev` builds from `master` across our CI services but those aren't made available to users as an official distribution.
+
+## Explanation
+
+A user who wants to use the latest improvements to Atom each day can go to atom.io, download the Atom Nightly release, and install it on their machine.  This release can be installed alongside Atom Stable and Atom Beta.
+
+Each night when there are new commits to Atom's `master` branch, a scheduled CI build creates a new Atom Nightly release with packages for Windows, macOS, and Linux.  These packages are automatically uploaded to a new GitHub release on the `atom/atom-nightly` repository using a nightly version based off of the current `dev` version in `master` (e.g. v1.29.0-dev.1 or v1.29.0-dev.20180601).
```

Comment:
One idea I'd had was to have this nightly build not just pull and build `master` directly, but pull `master`, then merge in any PRs with a special label applied, and build that instead.

This would give us a mechanism to live with big, invasive, risky changes, like major Electron version bumps or `apm` upgrades, without committing to shipping them in the next beta like merging them to master would.

Reply:
I love that idea, but then we would have to be careful about the potential for merge conflicts.

Reply:
Definitely a cool idea, but I think that would make things a lot more complicated.  Perhaps we could use feature flags for a lot of this?  If there are major changes that can't be feature flagged, we might be able to do one-off release builds (or channels) to get feedback.